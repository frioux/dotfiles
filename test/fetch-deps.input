#!/bin/bash
# FIXME: Add tests for DEP_{project}_URI with a relative URI.

dir="$PWD/test/,fetch-deps"
makefile="$dir/Makefile"
cache=".mduem/cache/Makefile.variables"

mkdir -p "$dir"
mkdir -p "$dir/$(dirname "$cache")"
cp "$cache" "$dir/$cache"
cat >"$makefile" <<END
DEPS := mduem
DEP_mduem_URI := $PWD
DEP_mduem_VERSION := test-fetch-deps

include $PWD/Makefile
END

cd "$dir"
make --always-make -f "$makefile" >/dev/null \
  fetch-deps

find -name '.git' -prune -or -print
( cd "$dir/.mduem/deps/mduem" && git describe --tags --always --dirty; )

rm -rf "$dir"

# __END__
