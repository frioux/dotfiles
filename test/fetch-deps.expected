.
./.mduem
./.mduem/cache
./.mduem/cache/Makefile.variables
./.mduem/deps
./.mduem/deps/,mduem
./.mduem/deps/,mduem.log
./.mduem/deps/mduem
./.mduem/deps/mduem/LICENSE
./.mduem/deps/mduem/Makefile
./.mduem/deps/mduem/README
./.mduem/deps/mduem/test
./.mduem/deps/mduem/test/all.expected
./.mduem/deps/mduem/test/all.input
./.mduem/deps/mduem/test/build.expected
./.mduem/deps/mduem/test/build.input
./.mduem/deps/mduem/test/clean-generated.expected
./.mduem/deps/mduem/test/clean-generated.input
./.mduem/deps/mduem/test/clean-junks.expected
./.mduem/deps/mduem/test/clean-junks.input
./.mduem/deps/mduem/test/clean.expected
./.mduem/deps/mduem/test/clean.input
./.mduem/deps/mduem/test/install-filter.expected
./.mduem/deps/mduem/test/install-filter.input
./.mduem/deps/mduem/test/install-generated.expected
./.mduem/deps/mduem/test/install-generated.input
./.mduem/deps/mduem/test/install-post-vim.expected
./.mduem/deps/mduem/test/install-post-vim.input
./.mduem/deps/mduem/test/install-post.expected
./.mduem/deps/mduem/test/install-post.input
./.mduem/deps/mduem/test/install-renaming.expected
./.mduem/deps/mduem/test/install-renaming.input
./.mduem/deps/mduem/test/install-static.expected
./.mduem/deps/mduem/test/install-static.input
./Makefile
test-fetch-deps
